### PR TITLE
Updates timeliness check for edit NOD date

### PR DESCRIPTION
**Pull request by bgantick**
### Description
Resolves unexpected behavior wherein users were able to set receipt date to before decision date for issues. This new check should override exemption status too.

### Acceptance Criteria
- [x] Code compiles correctly

### Testing Plan
1. Go to any AMA appeal as a user that's able to edit NOD date (judge for instance)
2. Try to set NOD date to before decision date of any issue that's attached to the appeal
3. Ensure you get a timeliness error message when you try to save the new NOD date

### CODE REVIEW

File: app/models/request_issue.rb [hunk 1]
```diff
@@ -625,6 +625,7 @@ def handle_legacy_issues!
 
   def timely_issue?(receipt_date)
     return true unless receipt_date && decision_date
+    return false if receipt_date < decision_date
```
Comment (meckila-navapbc) [inline app/models/request_issue.rb:628]: This line may need to be swapped with the next line to prevent

Comment (meckila-navapbc) [inline app/models/request_issue.rb:628]: Does this impact the timeliness of issues with exemptions?

File: client/app/queue/components/EditNodDateModal.jsx [hunk 2]
```diff
@@ -199,7 +199,7 @@ export const EditNodDateModal = ({ onCancel, onSubmit, nodDate, reason, showTime
         })}
       </ul>
 
-      <strong>Unaffected Issue(s)</strong>
+      { issues.unaffectedIssues.length > 0 && <strong>Unaffected Issue(s)</strong> }
```
Comment (meckila-navapbc) [inline client/app/queue/components/EditNodDateModal.jsx:202]: Good catch, all issues could be affected by the attempted update

File: app/models/request_issue.rb (hunk 1 above)
Comment (bgantick) [reply app/models/request_issue.rb:628]: It should, and this is intended. It shouldn't be possible to have an NOD date before a decision is made.

Comment (bgantick) [reply app/models/request_issue.rb:628]: (see other comment)

Comment (meckila-navapbc) [reply app/models/request_issue.rb:628]: Meant to delete this one!

**Review by meckila-navapbc — APPROVED** [review]

**Pull request merged**